### PR TITLE
Incubate Eggs

### PR DESCRIPTION
**Pull request by eddawang**
Currently works for one of the incubators. Not sure why but it's only finding one of the incubators and it doesn't seem to be the infinite one.

### CODE REVIEW

**Comment by destiny117** [comment]
cool, here's some input for you:

['inventory_item_data']['egg_incubators'] is an array of all incubators, you can check if its type by checking the value item_id: 901 is unlimited use, 902 are the limited use ones

furthermore you should make two seperate loops first, finding the incubators, then looping through the eggs and using them (maybe make an array of the eggs first to sort it?)

also you should make sure an incubator is not in use yet, for example `if "pokemon_id" in incubator` and make sure an egg is not being incubated atm, for example `if "egg_incubator_id" in incubator`

File: pgoapi/pgoapi.py [hunk 1]
```diff
@@ -731,6 +731,30 @@ def login(self, provider, username, password, cached=False):
...
         return True
+        
+    
+    def incubate_eggs(self, inventory_items=None):
+        if not inventory_items:
+            inventory_items = self.get_inventory().call()['responses']['GET_INVENTORY']['inventory_delta']['inventory_items']
+        incubators = None
+        for inventory_item in inventory_items:
+            #Finds incubators..for some reason only one is ever found
+            if "egg_incubators" in inventory_item['inventory_item_data']:
```
Comment (joychugh) [inline pgoapi/pgoapi.py]: please add egg info into inventory.py Thanks!

Comment (joychugh) [inline pgoapi/pgoapi.py]: also the incubator info.

**Comment by destiny117** [comment]
i dived a bit into it, and there's quite a bit more to incubating than you have written so far

i have now written a complete incubating method, including hatching, currently testing, will make a pull request later when everything works as expected

**Comment by j-e-k** [comment]
ok, will wait to see what @destiny117 has. Thanks guy for this, been on my backlog for far too long!
😄

**Comment by j-e-k** [comment]
I think we are going to go with #158 which builds further on this

**Pull request closed without merging**